sdk/core/src/CommonNetworkMember/BaseNetworkMember.ts

```diff
@@ -209,20 +206,21 @@ export abstract class BaseNetworkMember {
     password: string,
     keyOptions?: KeyOptions,
   ) {
+    const {
+      basicOptions: { registryUrl },
+      accessApiKey,
+    } = options
+    const api = new RegistryApiService({ registryUrl, accessApiKey, sdkVersion: extractSDKVersion() })
     const didMethod = options.otherOptions.didMethod || DEFAULT_DID_METHOD
-    const passwordBuffer = KeysService.normalizePassword(password)
-    const seedWithMethod = await generateFullSeed(dependencies.platformCryptographyTools, didMethod, keyOptions)
-    const encryptedSeed = await KeysService.encryptSeed(seedWithMethod, passwordBuffer)
-    const keysService = new KeysService(encryptedSeed, password)
-
-    const didDocumentService = DidDocumentService.createDidDocumentService(keysService)
-    const didDocument = await didDocumentService.buildDidDocument()
-    const did = didDocument.id
-    const didDocumentKeyId = didDocumentService.getKeyId()
-
-    await BaseNetworkMember._anchorDid(encryptedSeed, password, didDocument, 0, options)
+    const registeringService = new RegisteringService(
+      api,
+      didMethod,
+      dependencies.platformCryptographyTools,
+      password,
+      keyOptions,
+    )
 
-    return { did, didDocumentKeyId, encryptedSeed }
+    return registeringService.register()
```

Comment:
Again, there is no point in calling this a "service", just a method `export const register` would suffice.

Even if we want to call this a "service", it should be reusable. What point is there to have a "service" that only has one instance method that can only be called once?
For example, its constructor could accept `api` and `platformCryptographyTools`, while `register` method would accept `didMethod`, `password` and `keyOptions` (parameters specific to this registration)